crates/bytecode/src/iter.rs

```diff
@@ -6,66 +6,81 @@ use crate::{opcode, Bytecode, OpCode};
 /// without dealing with the immediate values that follow instructions.
 #[derive(Debug, Clone)]
 pub struct BytecodeIterator<'a> {
-    /// Reference to the underlying bytecode bytes
-    bytes: &'a [u8],
-    /// Current position in the bytecode
-    position: usize,
-    /// End position in the bytecode (to handle original length for legacy bytecode)
-    end: usize,
+    /// Start pointer of the bytecode. Only used to calculate [`position`](Self::position).
+    start: *const u8,
+    /// Iterator over the bytecode bytes.
+    bytes: core::slice::Iter<'a, u8>,
 }
 
 impl<'a> BytecodeIterator<'a> {
     /// Creates a new iterator from a bytecode reference.
+    #[inline]
     pub fn new(bytecode: &'a Bytecode) -> Self {
-        let bytes = bytecode.bytecode();
-        let end = match bytecode {
-            Bytecode::LegacyAnalyzed(analyzed) => analyzed.original_len(),
-            Bytecode::Eip7702(_) => 0,
-            _ => bytes.len(),
-        };
-
+        let bytes = bytecode.original_byte_slice();
```

Comment:
`original_byte_slice` bytes for Eip7702 would iterate over address, and for EOF it would iterate over header.

You should use `.bytecode()` for Lagacy/EOF and for Eip7702 a empty vec